### PR TITLE
[GSOC 1.2]removing instance of nsimplify from log _eval_nseries

### LINKED ISSUE #25175: nsimplify should not be used in series calculations
The use of `nsimplify` here leads to incorrect results:
https://github.com/sympy/sympy/blob/e2cadc140cc969fea038240a39961a66a2f3dd6d/sympy/functions/elementary/exponential.py#L1047-L1048

This series should just be all zero terms:
```python
In [3]: g1 = 2*acosh(1+2*z/3)-acosh(S(5)/3-S(8)/3/(z+4))

In [4]: g3 = 2*log(sqrt((z+4)/3)*(sqrt(z+3)+sqrt(z))**2/(2*sqrt(z+3)+sqrt(z)))

In [5]: s = (g1 - g3).series(z)

In [6]: s
Out[6]: 
     ⎛                    11  31   61  47⎞         ⎛     372  19  443  193                   17        
     ⎜                    ──  ──  ───  ──⎟         ⎜     ───  ──  ───  ───                   ──        
     ⎜                    16  32  192  48⎟         ⎜     485  97  485  485             7/59  59  9/59  
 9/2 ⎜  115963⋅√3   1000⋅2  ⋅3  ⋅5   ⋅7  ⎟    11/2 ⎜  2⋅2   ⋅3  ⋅5   ⋅7      19131876⋅2    ⋅3  ⋅5    ⋅7
z   ⋅⎜- ───────── + ─────────────────────⎟ + z    ⋅⎜- ──────────────────── + ──────────────────────────
     ⎝    746496            194481       ⎠         ⎝           45                    820654296875      

53             ⎞        
──             ⎟        
59             ⎟        
     4194103⋅√3⎟    ⎛ 6⎞
── + ──────────⎟ + O⎝z ⎠
      8211456  ⎠    
```
Internally `nsimplify` is called with:
```python
In [1]: nsimplify(-403*sqrt(3)/2737152)
Out[1]: 
                 17        53 
                 ──        ── 
           7/59  59  9/59  59 
-19131876⋅2    ⋅3  ⋅5    ⋅7   
──────────────────────────────
         820654296875 
```
This output from `nsimplify` is not equal to the input which could be considered a bug in `nsimplify` but then again `nsimplify` is only supposed to be approximate. Really the problem is that the series code should not use nsimplify at all. This is presumably added to handle floats but that should be done elsewhere.

The call to `nsimplify` was added in gh-23798 (CC @anutosh491).

### PR DESCRIPTION
**Pull request by arnabnandikgp**
Removed instance of  nsimplify from log eval_nseries
#### References to other Issues or PRs

Fixes #25175 

#### Brief description of what is fixed or changed
Removes instance of nsimplify from code and changed related test files.

#### Release Notes

* series
  * removed instance of nsimplify from log _eval_nseries

### CODE REVIEW

File: sympy/functions/elementary/tests/test_hyperbolic.py [hunk 1]
```diff
@@ -1540,3 +1544,10 @@ def test_issue_25847():
     #acsch
     assert acsch(sin(x)/x).as_leading_term(x) == log(1 + sqrt(2))
     raises(PoleError, lambda: acsch(exp(1/x)).as_leading_term(x))
+
+
+def test_issue_25175():
+    x = Symbol('x')
+    g1 = 2*acosh(1+2*x/3)-acosh(S(5)/3-S(8)/3/(x+4))
+    g2 = 2*log(sqrt((x+4)/3)*(sqrt(x+3)+sqrt(x))**2/(2*sqrt(x+3)+sqrt(x)))
```
Comment (anutosh491) [inline sympy/functions/elementary/tests/test_hyperbolic.py:1552]: Maybe take care of spacing here. I would write the following instead. 

`g1 = 2*acosh(1 + 2*x/3) - acosh(S(5)/3 - S(8)/3/(x + 4))`

**Review by anutosh491 — APPROVED** [review]
I like this change cause 
1) Using `nsimplify` was a very hackish solution that I used during my GSoC as back then I couldn't get `cancel` to work. We were using `cancel` on master before I introduced `nsimplify`  as can be seen [here ](https://github.com/sympy/sympy/pull/23798/files#diff-500498fb8c13a23d099bac93665453cd757898d898d70860b4b5a0c3b04c7e77)

I have verified the tests introduced too. @oscarbenjamin you can have a look !

File: sympy/functions/elementary/tests/test_hyperbolic.py (hunk 1 above)
Comment (anutosh491) [reply sympy/functions/elementary/tests/test_hyperbolic.py:1552]: Update g2 accordingly :)

Comment (arnabnandikgp) [reply sympy/functions/elementary/tests/test_hyperbolic.py]: Oops! changed them.

**Comment by oscarbenjamin** [comment]
Looks good to me.

**Comment by anutosh491** [comment]
Thank you for this, Merging !

**Pull request merged**